File: python_backend/custom/build_range_dict/sub_ids.py
from pprint import pprint
from helpers.step_funcs import *
from custom.build_range_dict.standard_ids import _standard_ids


def sub_ids(subs):
    """replace any of Bishop Knight or Queen in the game by assigning its id to the name of a custom piece """
    id_dict = {}
    for s in subs.keys():
        id_ = _standard_ids[subs[s]]
        id_dict[id_] = s
    return id_dict


if __name__ == "__main__":
    pprint(sub_ids({'Jester': 'Queen', 'Duke': 'Bishop'}))

File: app (1)/src/apiHelpers/idAssign/subIds.js
import {standardIds} from "./standardIds";


export function subIds(subs) {
    /**replace any of Bishop Knight or the.includes(Queen) game by assigning its id to the name of a custom piece.
       assign standard ids to custom piece names, resulting in a dict of form "standardId":"customPieceName"
    */
    var idDict = {}
    var standardPieceName, standardId;
    for (var customPieceName of Object.keys(subs)) {
        standardPieceName = subs[customPieceName]
        standardId = standardIds[standardPieceName]
        idDict[standardId] = customPieceName
    }
    return idDict
}

// module.exports = subIds;

/** for node.js
if (require.main === module) {
    console.log(subIds({'Jester': 'Queen', 'Duke': 'Bishop'}))

}
*/